java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -193,6 +193,10 @@ private static boolean hasOverloads(Symbol symbol) {
     return owner.isTypeSymbol() && calcOverloads((Symbol.TypeSymbol) owner, symbol.name()) > 1;
   }
 
+  private static boolean hasVarargArg(Symbol symbol) {
+    return symbol.isMethodSymbol() && ((Symbol.MethodSymbol) symbol).parameterTypes().stream().anyMatch(Type::isArray);
```

Comment:
Checking for any array is too much here. Varargs can only be placed on last parameter. There is also a dedicated method in JUtils for that:
```
return symbol.isMethodSymbol() && JUtils.isVarArgsMethod((Symbol.MethodSymbol) symbol);
```